opencensus/stats/exporters/prometheus_exporter.py

```diff
@@ -0,0 +1,339 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from prometheus_client import start_http_server
+from prometheus_client.core import CollectorRegistry
+from prometheus_client.core import GaugeMetricFamily
+from prometheus_client.core import CounterMetricFamily
+from prometheus_client.core import UntypedMetricFamily
+from prometheus_client.core import HistogramMetricFamily
+from prometheus_client.core import REGISTRY
+from opencensus.stats.exporters import base
+from opencensus.stats import aggregation_data as aggregation_data_module
+from opencensus.common.transports import sync
+
+
+class Options(object):
+    """ Options contains options for configuring the exporter.
+    The address can be empty as the prometheus client will
+    assume it's localhost
+    """
+    def __init__(self,
+                 namespace='',
+                 port=8000,
+                 address='',
+                 registry=CollectorRegistry()):
+        self._namespace = namespace
+        self._registry = registry
+        self._port = int(port)
+        self._address = address
+
+    @property
+    def registry(self):
+        """ Prometheus Collector Registry instance
+        """
+        return self._registry
+
+    @property
+    def namespace(self):
+        """ Prefix to be used with view name
+        """
+        return self._namespace
+
+    @property
+    def port(self):
+        """ Port number to listen
+        """
+        return self._port
+
+    @property
+    def address(self):
+        """ Endpoint address (default is localhost)
+        """
+        return self._address
+
+
+class Collector(object):
+    """ Collector represents the Prometheus Collector object
+    """
+    def __init__(self, options=Options(), view_data={}):
+        self._options = options
+        self._registry = options.registry
+        self._view_data = view_data
+        self._registered_views = {}
+
+    @property
+    def options(self):
+        """ Options to be used to configure the exporter
+        """
+        return self._options
+
+    @property
+    def registry(self):
+        """ Prometheus Collector Registry instance
+        """
+        return self._registry
+
+    @property
+    def view_data(self):
+        """ Map with all view data objects
+        that will be sent to Prometheus
+        """
+        return self._view_data
+
+    @property
+    def registered_views(self):
+        """ Map with all registered views
+        """
+        return self._registered_views
+
+    def register_view(self, view):
+        """ register_view will create the needed structure
```

Comment:
It is lower case because we start with the method name (as done in other classes). Do you prefer if we start with the description, like `Will create...`?